app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt

```diff
@@ -228,23 +229,65 @@ class SettingsActivity : PassphraseRequiredActionBarActivity() {
         }
     }
 
+    // Method to update the display name if possible
+    private fun handleDisplayNameChange(displayName: String?) {
+
+        // If we don't have a working network connection we'll decline the display name change and bail..
+        val haveNetworkConnection = NetworkUtils.haveValidNetworkConnection(this@SettingsActivity)
+        if (!haveNetworkConnection) {
+            binding.loader.isVisible = false
+            Log.w(TAG, "Declining attempt to change display name without network connection.")
+            Toast.makeText(this@SettingsActivity, R.string.profileErrorUpdate, Toast.LENGTH_LONG).show()
+            return
+        }
```

Comment:
Doesn't all of this also applies to the rest of the "update profile"?
Wouldn't we want to also show an error if there was no network but I was updating a picture (isUpdatingProfilePicture = true)?

Reply:
Good point - I'll make sure this applies to any profile picture changes as well.